model.py

```diff
@@ -166,6 +168,64 @@ class SoftplusLayer(tf.keras.layers.Layer):
     def call(self, inputs):
         return tf.math.log(1 + tf.exp(inputs))
 
+def cross_val_with_best_model(
+    model_fn: Callable,
```

Comment:
Could you specify more concretely what this needs to return? e.g. `Callable[SomeAbstractClass]`.